libs/metrics/in_out.py

```diff
@@ -207,11 +208,85 @@ def get_weekly_report(cls, entities: List[str], number_of_weeks: int = 0,
             ])
         return report
 
+    @classmethod
+    def _process_path(cls, boundary_line, trajectory_path, number_of_cuts=NUMBER_OF_PATH_SEGMENTS):
+        """
+        Verify if a trajectory goes over a boundary line
+        Args:
+            Two coordinates [x,y] are in 2-tuples [A,B]
+                Boundaries of the in/out line.
+                If someone crosses the line while having A to their right, they are going in the in direction (entering)
+                Crossing the line while having A to their left means they are going in the out direction (leaving)
+
+            trajectory_path: List of N 2-tuples (x,y)
+            That represents the trajectory of an object.
+
+        Returns:
+            (in, out) : tuple
+                 (1, 1) - if the object entered and left an equal number of times.
+                 (1, 0) - if the object entered (in)
+                 (0, 1) - if the object left (out)
+                 (0, 0) - if the object didn't cross the boundary.
+        """
+        if len(trajectory_path) < number_of_cuts:
+            number_of_cuts = len(trajectory_path)
 
-def fill_partially_empty_result(tuple_of_lists, default_value):
+        trajectory_steps = [trajectory_path[int(i)] for i in np.linspace(0, len(trajectory_path) - 1, number_of_cuts)]
+        trajectory_steps = zip(trajectory_steps, trajectory_steps[1:])
+        total_in, total_out = 0, 0
+
+        for trajectory in trajectory_steps:
+            path_in, path_out = check_line_cross(boundary_line, trajectory)
+            total_in += path_in
+            total_out += path_out
+
+        # Normalize in_out:
+        return (int(total_in >= total_out and total_in > 0), int(total_out >= total_in and total_out > 0))
+
+    @classmethod
+    def _process_hourly_segments(
+        cls, hourly_objects_logs, latest_estimated_occupancy, boundaries, # input
+        hour_in, hour_out, hour_balance, summary_report # output
+    ):
+        logger.info(summary_report)
```

Comment:
This does not seem necessary. Same below and if you think it is right to remove, also remove the imports etc.

Reply:
Yes, I thought I had them removed